### ISSUE TITLE
Limit of log(z) as z goes to 0 with z complex return '-oo' instead of 'zoo'

### ISSUE THREAD

**Issue by meindertsma** [sympy/sympy#24331]
```
import sympy as s
z = s.Symbol('z', complex=True)

str(s.limit(s.log(z), z, 0))
```
Output is `'-oo'`, it should be `-zoo`. It seems like the assumption is not handled (properly).

```
import sympy as s
z = s.Symbol('z', infinite=True)

str(s.limit(s.log(z), z, 0))
```
returns `'zoo'`.

**Comment by asmeurer** [comment]
`complex=True` doesn't really do what you are expecting. It's already an implicit assumption by most SymPy functions that symbols are complex numbers. All it really does beyond the default of no assumptions is tell it that z cannot be infinite, which shouldn't even matter for a limit near 0. 

`limit` currently only takes real limits. The default is a limit from the right, although there are discussions about changing it to be bidirectional (see https://github.com/sympy/sympy/pull/23516 and related issues). Adding support for complex limits is also something that isn't implemented yet (https://github.com/sympy/sympy/issues/8320).

The behavior with Symbol('z', infinite=True) is surprising. The computation itself is a bit nonsensical if you take the assumptions into account (z cannot approach 0 if it is infinite). But I don't think that limit should be looking at at assumptions at all, so something odd is happening. Perhaps at some point, `log(z)` itself is being evaluated as `zoo` since `z` is infinite. It doesn't automatically evaluate. The general iss of the complex limit issue can be considered a duplicate of those other issues, but the `infinite` thing should be investigated, so I'm going to leave this open fo rnow.

**Comment by jksuom** [comment]
> Perhaps at some point, log(z) itself is being evaluated as zoo since z is infinite.

That is what happens in a substitution here: https://github.com/sympy/sympy/blob/5ba43252efd64bcda5aa0f6bc27debdb6b0fefc2/sympy/series/gruntz.py#L728
```
>>> z = Symbol('z', infinite=True)
>>> log(z).subs(z, 1/z)
zoo
```
There is no need to change Gruntz's algorithm. Maybe `Limit.__new__` should check the "variable" `z` and raise an exception if it is actually not variable at all.

**Comment by meindertsma** [comment]
Hmm, I didn't see that the default limit is from the right, the documentation should really be clearer about that. 
I also want to note that `s.log(0)` does return complex infinity.

**Comment by jksuom** [comment]
> I didn't see that the default limit is from the right,

> In [101]: Limit?
Init signature: Limit(e, z, z0, dir='+')

**Comment by meindertsma** [comment]
Didn't notice that first, but the docstring says `Computes the limit of ``e(z)`` at the point ``z0``.`. I think this should be changed to `Computes the right-side limit of ``e(z)`` at the point ``z0``.` Should I make a pull request? Or wait for https://github.com/sympy/sympy/pull/23516 to be resolved?

**Comment by asmeurer** [comment]
The change to bidirectional might not happen any time soon. It's something we've wanted for a long time but there are a lot of difficulties involved. So a documentation improvement would definitely be welcome.

**Comment by eagleoflqj** [comment]
> Or wait for #23516 to be resolved?

The only reason I haven't finished #23516 is there is no consensus reached yet. I can do the chores to add all `dir='+'` in sympy code but I should first get a green light.

**Comment by faze-geek** [comment]
> > Perhaps at some point, log(z) itself is being evaluated as zoo since z is infinite.

Yes indeed, `infinite = True` totally messes up these evaluations -
```
>>> z = Symbol('z')
>>> str(limit(log(z),z,oo))
'oo'
>>> str(limit(log(z),z,0))
'-oo'
>>> str(limit(log(z),z,1))
'0'
>>> str(limit(log(z),z,2))
'log(2)'
```

```
>>> z = Symbol('z',infinite = True)
>>> str(limit(log(z),z,oo))
"Limit(log(z), z, oo, dir='-')"
>>> str(limit(log(z),z,0))
'zoo'
>>> str(limit(log(z),z,1))
'0'
>>> str(limit(log(z),z,2))
'nan'

```

I somewhat agree with @asmeurer that limit should not look at the assumptions in the first place (it will avoid weird limits like an `infinite` symbol tending to `0`). We can do something like -

```
+        if z has any assumptions :
+            from sympy.core import Dummy
+            x = Dummy('x', real=True)
+            e = e.subs(z, x)
+            z = x
+
```

If assumptions are used, I would also expect consistency -
```
>>> z = Symbol('z',integer = True)
>>> str(limit(log(z),z,0.33333333333))   # How can a predefined integer tend to a floating value like here 
'-1.09861228867811'
```

> There is no need to change Gruntz's algorithm. Maybe `Limit.__new__` should check the "variable" `z` and raise an exception if it is actually not variable at all.

I was looking into this but could not clearly get it. Hi @jksuom, could you kindly explain what is meant by raising an exception if z is not a variable. Is it supposed to mean that if `z` evaluates to `zoo/-zoo/nan` somewhere when an assumption is used, then an error must be raised?

**Comment by jksuom** [comment]
> what is meant by raising an exception if z is not a variable.

I think that a "variable" should be able to "vary", i.e., take all values in a neighborhood of the limit point.. If a symbol has `zero=True` or `infinite=True`, then it cannot be a limit variable. If a symbol has `integer=True` and the limit point is not `+/-oo`, then it should not be used in a limit. There are also other impossible combinations though they will probably not appear in practice. These could be checked in `Limit.__new__`.
